src/Moryx.Runtime/Modules/ServerModuleBase.cs

```diff
@@ -115,17 +122,24 @@ void IServerModuleStateContext.Initialize()
             Notifications.Clear();
         }
 
-        void IServerModule.Start()
+        async Task IServerModule.StartAsync()
         {
-            lock (_stateLock)
-                _state.Start();
+            await _stateLockSemaphore.WaitAsync();
```

Comment:
MAybe we can already settle on a recommendation whether to use `SemaphoreSlim` or `Lock` for state locks 🤔

FYI: @seveneleven

Reply:
again - lock is not built to use with async.